src/middlewared/middlewared/plugins/notifier.py

```diff
@@ -135,6 +135,14 @@ def directoryservice(self, name):
             pass
         return user
 
+    async def get_group_object(self, groupname):
```

Comment:
This should be a method, not a coroutine to avoid middleware lockup.
Please also fix get_user_object.